### PR TITLE
Implement Copy email List Section

### LINKED ISSUE #132: [Feature] Implement Copy email List Section
In this new section, a user will be able to select a cabin from a dropdown and hit the copy button icon to copy a string of emails associated with that cabin! 

### Your job is:

- [x] Implement the design [here](https://www.figma.com/file/Ur8DMCmxLx7WMKqHoEChR3/Cabin-Sorting?type=design&node-id=102-632&mode=design&t=wWOnY8zbz8f2g9Ue-0) using mui components entirely (dropdown title should be "Cabin")
- [x] For functionality, use dummy data to copy to the user's clipboard
- [x] Once copied, the copy icon button should turn into the check mark icon (non-clickable)
- [x] Once copied, a Snackbar should appear in the top right stating "Copied to Clipboard"
- [x] Check icon should be green
- [x] If the user goes to select another cabin, the check mark icon should turn back into the copy icon button

### PR DESCRIPTION
**Pull request by kzeng24**
Fixes #132 

- Implemented the design [here](https://www.figma.com/file/Ur8DMCmxLx7WMKqHoEChR3/Cabin-Sorting?type=design&node-id=102-632&mode=design&t=wWOnY8zbz8f2g9Ue-0) 
- dropdown showing all cabin options 
- clickable copy icon will appear next to the dropdown once a valid cabin is selected, that copies the emails from that cabin
- checkmark icon will appear next to the dropdown after the user copies content
- a snackbar will appear at the bottom left after cabin content is copied
- snackbar/simpleSnackBar.tsx can take in any string message and horizontal/vertical positioning
- made sure all icons come from material ui

Screenshots:
Screen Shot 2023-07-04 at 10 39 28 PM
Screen Shot 2023-07-04 at 10 39 48 PM
Screen Shot 2023-07-04 at 10 40 03 PM
List of emails copied from Cabin 2:
Screen Shot 2023-07-04 at 10 40 34 PM

### CODE REVIEW

**Review by zhangcry — CHANGES_REQUESTED** [review]
Great work!!! One thing I was wondering about was the use of React.Fragment. To be clear, I didn't know anything about it before doing this review. I saw on some resources that you could use <> ... </> instead. Is there a reason you chose to write out the name rather than using the shorthand?

File: ui/components/templateDropdown/cabinDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          </span>)
+          : (
+          <span
+            style={{
+              marginLeft: '1em',
+              position: 'absolute',
+              top: '21%'
+            }}
+            onClick={() => {
+              navigator.clipboard.writeText(selectedItem)
```
Comment (mike10911) [inline ui/components/templateDropdown/cabinDropdown.tsx:78]: We'd like to return the string of emails associated with this cabin, not the cabin name. Find a way to pass these down!

File: ui/components/templateDropdown/cabinDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import TopRightSnackBar from '../snackBar/topRightSnackBar'
+
+type CabinDropdownProps = {
+  items: string[];
+};
+
+export default function CabinDropdown ({ items }: CabinDropdownProps) {
+  const [selectedItem, setSelectedItem] = React.useState('')
+  // if false, the copy icon will appear
+  // if true, the checkmark will appear
```
Comment (mike10911) [inline ui/components/templateDropdown/cabinDropdown.tsx:22]: remove these comments

File: ui/components/templateDropdown/cabinDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import { MdContentCopy } from 'react-icons/md'
+import { IoMdCheckmark } from 'react-icons/io'
+import TopRightSnackBar from '../snackBar/topRightSnackBar'
+
+type CabinDropdownProps = {
+  items: string[];
+};
+
+export default function CabinDropdown ({ items }: CabinDropdownProps) {
+  const [selectedItem, setSelectedItem] = React.useState('')
```
Comment (mike10911) [inline ui/components/templateDropdown/cabinDropdown.tsx:20]: Let's call this selectedCabin to be more precise

File: ui/components/snackBar/topRightSnackBar.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import * as React from 'react'
```
Comment (mike10911) [inline ui/components/snackBar/topRightSnackBar.tsx:1]: Let's abstract this component to be any position snack bar, receiving message, and position props.

**Review by mike10911 — CHANGES_REQUESTED** [review]
This is soo good guys!!! A few changes below:
- Update the mui select component to have a label + helper text (https://codesandbox.io/s/24jqtk?file=/demo.tsx)
- "Cabin" should not be a option when selecting, rather it should be the label/helper text
- For the copy/check icons use mui icons and remove react icons as a dependency
- I am not realizing that putting the snackbar in the top right covers the signout button. Let move it to the bottom left

**Review by mike10911 — CHANGES_REQUESTED** [review]
Also plz create a proper PR comment detailing changes, you should NEVER just copy-paste the issue details. As it is a frontend ticket, be sure to include screenshots/video

File: ui/components/templateDropdown/selectedCabin.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+          >
+            <span color="#5e5d5d">
+              <ContentCopyIcon fontSize='medium'/>
+            </span>
+          </span>
+            )}
+
+        {openSnackBar && (
+          <SimpleSnackBar
+            message={'Copied to Clipboard'}
```
Comment (mike10911) [inline ui/components/templateDropdown/selectedCabin.tsx:93]: this should just be a string

File: ui/components/templateDropdown/selectedCabin.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+          : (
+          <span
+            style={{
+              marginLeft: '1em',
+              position: 'absolute',
+              top: '21%'
+            }}
+            onClick={() => {
+              navigator.clipboard.writeText(
+                cabinValues[selectedItem].toString()
```
Comment (mike10911) [inline ui/components/templateDropdown/selectedCabin.tsx:79]: remove toString

**Review by mike10911 — APPROVED** [review]
OMG, GREAT JOB WITH ALL THE CHANGES!!! Approving with a couple small changes.

**Review by zhangcry — APPROVED** [review]
great work!!!

**Review by mike10911 — CHANGES_REQUESTED** [review]
Sorry, I missed something! The copy icon should only appear when the user selects a cabin, to ensure we don't copy down undefined.

**Review by mike10911 — CHANGES_REQUESTED** [review]
When you addressed merge conflicts, the link to the cabin page was removed!

**Review by mike10911 — APPROVED** [review]
LGTM!!!!

**Pull request merged**